extras/mmu.py

```diff
@@ -2006,26 +2006,52 @@ def _get_print_status(self):
             self._log_trace("Determined print status as: %s from %s" % (print_status, source))
             return print_status
 
+    def _wait_for_target_temp(self):
+        target_temp = self.printer.lookup_object(self.extruder_name).heater.target_temp
+        current_temp = self.printer.lookup_object(self.extruder_name).get_status(0)['temperature']
+
+        if abs(target_temp - current_temp) < 1:
+            self._log_debug("Waiting for extruder to reach target temperature (%.1f)" % target_temp)
+            self.gcode.run_script_from_command("TEMPERATURE_WAIT SENSOR=extruder MINIMUM=%.1f MAXIMUM=%.1f" % (target_temp - 1, target_temp + 1))
+
     # Ensure we are above desired or min temperature and that target is set
-    def _set_above_min_temp(self, target_temp=-1):
+    def _set_above_min_temp(self, target_temp_override=-1):
         extruder_heater = self.printer.lookup_object(self.extruder_name).heater
         current_temp = self.printer.lookup_object(self.extruder_name).get_status(0)['temperature']
+        target_temp = max(target_temp_override, extruder_heater.target_temp, self.min_temp_extruder)
+
+        # During a print, we likely want to defer to the slicer for temperature since doing so
+        # will prevent the following issues:
+        #   1. When printing with different temperature filaments, this prevents waiting for the higher
+        #      temperature before unloading the lower-temperature filament
+        #   2. Prevents setting the wrong temp if your `min_temp_extruder` is higher than the temp of the filament
+        #
+        # This does mean that we're trusting the slicer to have set the correct temperature, but that's a reasonable
+        # assumption and we do maintain the safeguard of Klipper's `can_extrude` flag
+        if (self._is_in_print() or self._is_in_pause()) and extruder_heater.can_extrude:
+            if target_temp_override == -1:
+                self._log_info("Deferring to slicer for extruder temperature (%.1f)" % extruder_heater.target_temp)
+            else:
```

Comment:
I've just realized this won't catch the case where a heater turns off during a pause. Even if `target_temp_override` is set, it'll never get to that point because `can_extrude` is false. Will update